Newsletters: add 320 (2024-09-13) (#1881)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2024-09-13-newsletter.md

---
title: 'Bitcoin Optech Newsletter #320'
permalink: /en/newsletters/2024/09/13/
name: 2024-09-13-newsletter
slug: 2024-09-13-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME

## News

- **Mutation testing for Bitcoin Core:** Bruno Garcia [posted][] to
  Delving Bitcoin to announce a [tool][mutation-core] that would
  automatically modify (mutate) code changed in a PR or commit to

>   determine whether they will cause existing tests to fail.  Anytime

Reviewer: ```suggestion
  determine whether it will cause existing tests to fail.  Anytime
```

>   random changes to code doesn't produce a failure indicates that test

Reviewer 2: ```suggestion
  random changes to code don't produce a failure, it indicates that test
```

  coverage may be incomplete.  Garcia's automatic mutation tool ignores
  code comments and other lines of code that wouldn't be expected to
  produce changes.

- **DLC-based loan contract execution:** Shehzan Maredia

>   [posted][maredia post] Delving Bitcoin to announced [Lava Loans][], a

Reviewer 3: ```suggestion
  [posted][maredia post] Delving Bitcoin to announce [Lava Loans][], a
```

Reviewer 4: Also missing “to”

```suggestion
  [posted][maredia post] to Delving Bitcoin to announce [Lava Loans][], a
```

  The `bdk_wallet` crate "is the first version to offer a stable 1.0.0 API."

- [Bitcoin Core 28.0rc1][] is a release candidate for the next major
  version of the predominant full node implementation.  A [testing
  guide][bcc testing] is available.

<!-- FIXME:harding to update Thursday -->

## Notable code and documentation changes

_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

_Note: the commits to Bitcoin Core mentioned below apply to its master
development branch, so those changes will likely not be released until
about six months after the release of the upcoming version 28._

- [Bitcoin Core #30509][] adds an `-ipcbind` option to `bitcoin-node` to allow

>   other processes  to connect to and control the node via a unix socket. In

Reviewer 2: ```suggestion
  other processes to connect to and control the node via a unix socket. In
```

  combination with the upcoming PR [Bitcoin Core #30510][], this will allow an
  external [Stratum v2][topic pooled mining] mining service to create, manage,
  and submit block templates. This is part of the Bitcoin Core [multiprocess
  project][multiprocess project]. See Newsletters [#99][news99 multi] and
  [#147][news147 multi].

- [Bitcoin Core #29605][] changes peer discovery to prioritize peers from the

>   local address manager over fetching from seed nodes, to reduce the influence

Reviewer 2: I think it could be beneficial to have an additional sentence explaining how, before this PR, seed nodes can influence peer selection and cause unnecessary information sharing

v28-rc-testing] was a review club meeting that did not review a
particular PR, but rather was a group testing effort.

Before each [major Bitcoin Core release][], extensive testing by the
community is considered essential. For this reason, a volunteer writes a
testing guide for a [release candidate][] so that as many people as
possible can productively test without having to independently ascertain
what's new or changed in the release, and reinvent the various setup
steps to test these features or changes.

Testing can be difficult because when one encounters unexpected
behavior, it's often unclear if it's due to an actual bug or if the
tester is making a mistake. It wastes developers' time to report bugs to
them that aren't real bugs. To mitigate these problems and promote
testing efforts, a Review Club meeting is held for a particular release
candidate, in this instance, 28.0rc1.

The [28.0 release candidate testing guide][28.0 testing] was written by
rkrux, who also hosted the review club meeting.

Attendees were also encouraged to get testing ideas by reading the [28.0
release notes][].

This review club covered the introduction of [testnet4][topic testnet]

> ([Bitcoin Core #29775][]), [TRUC (v3) transactions][topic v3

Reviewer 2: I dont have a strong preference on it, but we could link to the Optech PR summaries in this section (vs linking to the PR itself)

[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

_Note: the commits to Bitcoin Core mentioned below apply to its master
development branch, so those changes will likely not be released until
about six months after the release of the upcoming version 28._

- [Bitcoin Core #30509][] adds an `-ipcbind` option to `bitcoin-node` to allow
  other processes to connect to and control the node via a unix socket. In
  combination with the upcoming PR [Bitcoin Core #30510][], this will allow an
  external [Stratum v2][topic pooled mining] mining service to create, manage,
  and submit block templates. This is part of the Bitcoin Core [multiprocess
  project][multiprocess project]. See Newsletters [#99][news99 multi] and
  [#147][news147 multi].

- [Bitcoin Core #29605][] changes peer discovery to prioritize peers from the
  local address manager over fetching from seed nodes, to reduce the influence
  of the latter on peer selection and reduce unnecessary information sharing.
  By default, seed nodes are a backup in case all DNS seeds are
  unreachable (which is very uncommon on mainnet); however users of test

>   networks or customized nodes many manually add seed nodes to find

Reviewer 5: ```suggestion
  networks or customized nodes may manually add seed nodes to find
```

  similarly configured nodes.  Before this PR, adding a seed node would
  result in it being queried for new addresses almost every time the
  node was started, potentially allowing it to influence peer selection
  and to only recommend peers that shared data with it.  With this PR,
  only when the address manager is empty, or after a period of unsuccessful
  address attempts, will seed nodes be added to the address fetch queue in
  random order. See Newsletter [#301][news301 seednode] for more on seed nodes.

{% assign four_days_after_posting = page.date | date: "%s" | plus: 345600 | date: "%Y-%m-%d 14:30" %}
{% include snippets/recap-ad.md when=four_days_after_posting %}
{% include references.md %}
{% include linkers/issues.md v=2 issues="30509,29605,30510,29775,28948,28984,27307,30493,30352,29347" %}

> [LND v0.18.3-beta.rc2]: https://github.com/lightningnetwork/lnd/releases/tag/v0.18.3-beta.rc2

Reviewer 4: ```suggestion
[LND v0.18.3-beta]: https://github.com/lightningnetwork/lnd/releases/tag/v0.18.3-beta
```

Reviewer 2: Thanks @jirijakes , took this
